### DISCUSSION TITLE
Statamic describes itself as open source while its not

### DISCUSSION THREAD

**Discussion by Simple-Minute** [statamic/cms#10706] [General]
Hi,
according to this Reddit post Statamic describes itself as open source, but it's actual licence is not open source.
https://www.reddit.com/r/opensource/comments/1f1xhon/is_statamic_open_source/

I think this really doesnt look good and looks like false advertising.
Statamic should either remove all instances where it calls itself "open source" and use instead correct term "source available" or it would need to change it's licence to be actually open source.

But currently it sadly calls itself falsely open source even if it's not.
I think this is just honest mistake that can be easily corrected.

Thanks.

**Comment by jackmcdade** [comment]
Open Source does not mean it needs to be free or use any one specific license. People always debate the definition, but ultimately open source means the source is open and avaialble and you're allowed to modify and release what you use it for according its the license.

https://www.reddit.com/r/opensource/comments/k7kj4m/can_something_paid_be_called_open_source/

**Reply by ssddanbrown** [reply]
> Open Source does not mean it needs to be free or use any one specific license. 

True, but charging for the software is not the concern raised, nor is there a request to use a specific license. Many people consider open source to be defined by the [OSI stewarded open source definition](https://opensource.org/osd), and a wider net of people will generally consider open source to assure open use, modification & distribution; which your custom license here prevents all of.

I respect your right to choose a license that fits your needs, but I think labelling a project under the statamic license as open source could very much be misleading to many, especially since the license sets quite substantial limits to common freedoms.

If you're interested in why I (and others) may be passionate about this I've written some relevant blog posts:
https://danb.me/blog/why-open-source-term-is-important/
https://danb.me/blog/open-source-available-distinction/

**Reply by Simple-Minute** [reply]
Hi @jackmcdade, so if I understand this correctly Statamic is open source according to you. So I can freely fork it, rename it and distribute or sell it myself? Basically, I can take this repo, make my own modifications to your code and distribute it myself?

**Reply by dannyuk1982** [reply]
> Hi @jackmcdade, so if I understand this correctly Statamic is open source according to you. So I can freely fork it, rename it and distribute or sell it myself? Basically, I can take this repo, make my own modifications to your code and distribute it myself?

The [license](https://statamic.com/license) states 

"you may not:
- Reproduce, distribute, or transfer the Software, or portions thereof, to any third party.
- Sell, rent, lease, assign, or sublet the Software or portions thereof."

**Reply by jackmcdade** [reply]
> Hi @jackmcdade, so if I understand this correctly Statamic is open source according to you. So I can freely fork it, rename it and distribute or sell it myself? Basically, I can take this repo, make my own modifications to your code and distribute it myself?

No you cannot. If everyone is really bothered by the term I can consider changing it but it’s all semantics. Source available does not have the same implications as open source in terms of open to contribution and collaboration, hence why we call it open source because the source is open. And the solo version is free to use no charge.

**Reply by Simple-Minute** [reply]
Thanks @dannyuk1982. But that means it's not open source.

@jackmcdade I understand it's confusing term. That's why I asked about it on Reddit in the first place. I am just trying to understand what is what.

I myself won't force it more than that. I understand now what I wanted to know. Thanks.

**Reply by ssddanbrown** [reply]
> hence why we call it open source because the source is open

Only open to viewing though, not open use, not open to modification, not open to redistribution (due to limits on each set in the license).

**Reply by JonKaric** [reply]
Maybe the Open Source Initiative needs to come out with an appropiately titled licence then? When I hear the term 'open source', I interpret that as the source is open for viewing. I can then read the licence and determine what I can and can't do with that software. I find it kinda silly that an initiative has sorta hijacked two words and given it it's own definition.

My local corner shop is open, but I can't just take what I want and leave.

**Reply by Simple-Minute** [reply]
@JonKaric When you mentioned "hijacking" the term it made me wonder what is even history of the term. Luckily there is nice summary on Wikipedia https://en.wikipedia.org/wiki/Open_source#Open_source_as_a_term
Also here https://en.wikipedia.org/wiki/The_Open_Source_Definition

And from reading this is looks like founders of Open Source Initiative actually co-created and popularized the term "open source". So they definitely didnt "hijack" it as you say. They were one of the original authors and were responsible for establishing meaning behind it. So what Open Source Initiative says "open source" means is by definition what it means.

**Reply by ssddanbrown** [reply]
More importantly in my opinion, there have been decades of people reinforcing that original definition by sharing work that adheres to that definition under the label of open source. It's that, and the OSI's stewardship, that have helped to establish it was a ubiquitous term. Work shared as open source while not providing those freedoms works to erode that, which is why folks like me are interested and put effort to push back against such use.

With any shorter term representing a larger cause, there may be some effort to understand the meaning beyond assumptions. "Free speech" is a good equivalent, with many misunderstand the actual meaning & movements behind that term based on their assumption (thinking it means they're free to say anything at any time without consequences), but the term to its proper meaning, and the fights/groups/causes/history behind it, are still important.

**Reply by JonKaric** [reply]
I think it's a term that clearly causes confusion, we've seen it all over this post. The OP concluded they can do anything with Statamic because it's labled as open source.

Seems like the term you're pushing is causing people to assume you can do absolutely anything with what you call open source software, and completely ignore licences which are the single source of truth for what you can do with the software.

Eg, one of the OSI approved* licences includes:

> You may modify your copy or copies of NetHack or any portion of it, and copy and distribute such modifications under the terms of Paragraph 1 above (including distributing this License Agreement), provided that you also do the following: a) cause the modified files to carry prominent notices stating that you changed the files and the date of any change; ...

> You may not copy, sublicense, distribute or transfer NetHack except as expressly provided under this License Agreement. 

Do you think people who just take your term at face value are gonna listen to this licence? I wouldn't think so. They will hear that you can "freely fork it, rename it and distribute or sell it myself?" and do what they want with it.

**Reply by ssddanbrown** [reply]
> The OP concluded they can do anything with Statamic because it's labled as open source.

I didn't see them state they expect to be able to do anything.

> Seems like the term you're pushing is causing people to assume you can do absolutely anything with what you call open source software

I don't see why. Open source provides open use, modification & distribution but nowhere is it said this allows "absolutely anything". It comes back to the [OSD](https://opensource.org/osd) which sets good rules as to what would be considered open source.

> Do you think people who just take your term at face value are gonna listen to this licence?

No, but we're specifically providing input to ask for "open source" to be considered to an existing established definition, not face value.

> They will hear that you can "freely fork it, rename it and distribute or sell it myself?"

They can totally do all that using that license example provided, as long as you also follow the requirements of the license.

**Reply by Simple-Minute** [reply]
Just for the record what I meant. I know there are basically two groups of open source licenses. Permissible like BSD or MIT where I can do almost anything. I can for example put it into proprietary project where nobody can get the source code. But I cant change the licence of the original code and I have to keep licence notice file or other such things. And then there is GPL group where I am forced to allow others the same freedoms it granted me, so no closing and I have to make source code available.
I know even every open source licence is kinda specific and different but the basics should be the same.

And because of those basics I expected I can take any open source code and "freely fork it, rename it and distribute or sell it myself". 
When I read the mentioned NetHack licence it seems it would allow me exactly what I described. It allows me to fork it, modify it, and sell it. I just need to also give source code to anyone I sell it to with the same licence.

So that's why I was confused. When I read something is "open source" I expect it's licence should be at least basically similar to other open source licenses.

**Comment by jackmcdade** [comment]
I don’t want this kind of confusion or arguments to be an ongoing thing. I’ll update our verbiage when I get a chance to make it more clear.

**Comment by viktorix** [comment]
Was trying to learn more about Statamic and was looking at the license. 

> Open Source does not mean it needs to be free or use any one specific license.

While it doesn't have to be free, if you want open source community to consider Statamic open source you should use a license that complies with Open Source Definition from OSI. List of licenses is here:
https://opensource.org/licenses

As I see it now, Statamic is [source available](https://en.m.wikipedia.org/wiki/Source-available_software) and not open source. Just like Sentry, Redis, Terraform, etc.

There's nothing wrong with that, but it's important to use the correct language and not try to [open wash](https://en.m.wikipedia.org/wiki/Openwashing) it.

**Comment by jackmcdade** [comment]
I would say we most closely resemble the [Open Core](https://en.m.wikipedia.org/wiki/Open-core_model) model, which is also argued whether it's truly open source or not. 🤔

**Reply by ssddanbrown** [reply]
Not really IMO, open core is where there's a core open source offering, with an alternative offering based on that core that has proprietary (non-open-source) additions/parts. Statamic is not open source at all and does not fit here. Source available fits.

**Reply by JonKaric** [reply]
@ssddanbrown Is there any wording that would be acceptable, that isn't `Source available`? I'm releasing something similar in licence to statamic soon, but don't want to use it as it implies that it's just available and nothing more. At least to me?

**Reply by ssddanbrown** [reply]
@JonKaric There's not too many other well established terms. Here's a few others within that range, although they are backed with definitions, so I can't validate if your software would fit them without knowing your license:

- [Fair source](https://fair.io/about/) - Lead by Sentry
- [Source first](https://sourcefirst.com/) - Lead by Futo
- [Fair code](https://faircode.io/) - Lead by n8n

**Comment by ssddanbrown** [comment]
Just an update that the project is still marketing itself as open source.
I've noted references to mentions on my site here: https://isitreallyfoss.com/projects/statamic/

**Reply by dannyuk1982** [reply]
Does Statamic claim to be FOSS?

**Reply by ssddanbrown** [reply]
It claims to be open source, as per my referenced link:

> The project advertises itself as open source in many locations, including in [their GitHub profile](https://github.com/statamic/.github/blob/cd827ca549520c16ce4e6581bf001f9f1a3212cd/profile/README.md#L3), on [their website](https://statamic.com/laravel-cms), and in [their licensing documentation](https://statamic.dev/licensing).

A couple of those mentions do specifically reference it as `Free and Open Source`.

**Reply by Amarjit** [reply]
They probably still call it “open source” because it gives the project a positive image and shows it’s transparent and accessible.

**Comment by vitalijalbu** [comment]
hope it will be one day... 😆

**Comment by Simple-Minute** [comment]
Honestly, writing on [licence page](https://statamic.dev/getting-started/licensing) that "**Statamic Core is free and open source and can be used for anything you wish**" really sounds like giving right to use, share, modify, and distribute the software as per definition of [FOSS](https://en.wikipedia.org/wiki/Free_and_open-source_software).
I just imagine the situation.

Someone forks, modifies and distributes [this Statamic repo](https://github.com/statamic/cms), which contains Statamic Core as written "This repository contains the code for the core Statamic Composer package"

Then Statamic authors sue the dev behind the fork and the dev defends themselves like "Well, Statamic literally told me the core is **free and open source** and that **I can use it for anything I wish**. They gave me permission.

**Comment by jackmcdade** [comment]
A docs page and the actual Terms included in the project are different, but point taken regardless. I’ve tweaked the verbiage there.

**Reply by Amarjit** [reply]
Can you link the commit?

**Reply by marcorieser** [reply]
Probably that one: https://github.com/statamic/docs/commit/e2b9ccf216eea31d8c01c769be90ff1867bb1dfa
